tests/tests/build

```diff
@@ -20,8 +20,38 @@ export CARGO_TARGET_DIR=$OBJDIR
 cargo clean
 cargo build
 
-# Normalize the save-analysis paths.
+## Re-map files dynamically generated by build.rs to a consistent path.
+#
+# build.rs uses an OUT_DIR which, if we don't transform it, will result in the
+# generated rust file living at a path that looks like
+# "$OBJDIR/debug/build/files-$HASH/out/generated.rs" where $HASH is some
+# (unpredictable for our purposes) hash.  Note that there will be multiple
+# directories matching a "$OBJDIR/debug/build/files-*" spec.
+#
+# Because searchfox doesn't generate permalinks to __GENERATED__ files, it ends
+# up being on the build scripts to make sure paths are as normalized as
+# possible.  We're taking on this responsibility here. See
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1646069 for more context about
+# this decision.
+
+# Identify the generated source path.
+BUILD_SCRIPT_OUT_FILE=$(ls $OBJDIR/debug/build/files-*/out/generated.rs)
+BUILD_SCRIPT_OUT_DIR=$(dirname "$BUILD_SCRIPT_OUT_FILE")
+# Create a symlink to normalize this directory so that we can expose it as
+# "__BUILD_SCRIPT_OUTDIR__".
+BUILD_SCRIPT_ALIAS="__BUILD_SCRIPT_OUTDIR__"
+BUILD_SCRIPT_OUT_ALIASED_PATH="${OBJDIR}/${BUILD_SCRIPT_ALIAS}"
+ln -sf "$BUILD_SCRIPT_OUT_DIR" "$BUILD_SCRIPT_OUT_ALIASED_PATH"
+
+## Normalize the save-analysis paths.
+# Normalize the source root off to make source paths relative.
 NORMALIZE_EXPR='s#/vagrant/tests/tests/files/##g'
+# Normalize the dynamic build script output directory to our new alias.  The
+# OBJDIR prefix removal will be handed by the next normalization.
```

Comment:
typo: handled

Reply:
Argh, thanks.  My keyboard.io keyswitches really need to be ultrasonically cleaned... thanks to a bad manufacturing process and imperfect software fixes they randomly drop some letters and double some other letters... it keeps me on my toes, but obviously I miss some things... also, sometimes it's my fault ;)